core/amber/src/main/python/core/architecture/packaging/output_manager.py

```diff
@@ -179,17 +179,14 @@ def add_partitioning(self, tag: PhysicalLink, partitioning: Partitioning) -> Non
         the_partitioning = get_one_of(partitioning)
         logger.debug(f"adding {the_partitioning}")
         for channel_id in the_partitioning.channels:
-            # Explicitly set is_control to trigger lazy computation.
-            # If not set, it may be computed at different times,
-            # causing hash inconsistencies.
-            channel_id.is_control = False
-            self._channels[channel_id] = Channel()
+            if channel_id.from_worker_id.name == self.worker_id:
+                # Explicitly set is_control to trigger lazy computation.
+                # If not set, it may be computed at different times,
+                # causing hash inconsistencies.
+                channel_id.is_control = False
+                self._channels[channel_id] = Channel()
         partitioner = self._partitioning_to_partitioner[type(the_partitioning)]
-        self._partitioners[tag] = (
-            partitioner(the_partitioning)
-            if partitioner != OneToOnePartitioner
-            else partitioner(the_partitioning, self.worker_id)
-        )
+        self._partitioners[tag] = partitioner(the_partitioning, self.worker_id)
```

Comment:
Also update the description to include this change, and explain why this issue doesn’t occur within each partitioner

Reply:
This change is unnecessary. Removed.